test/test_fp16.c

```diff
@@ -0,0 +1,154 @@
+         wasm_f16x8_extract_lane(v1, 3),
+         wasm_f16x8_extract_lane(v1, 4),
+         wasm_f16x8_extract_lane(v1, 5),
+         wasm_f16x8_extract_lane(v1, 6),
+         wasm_f16x8_extract_lane(v1, 7));
+}
+
+void assert_lanes_eq_uint16(v128_t v,
+                            uint16_t l0,
+                            uint16_t l1,
+                            uint16_t l2,
+                            uint16_t l3,
+                            uint16_t l4,
+                            uint16_t l5,
+                            uint16_t l6,
+                            uint16_t l7) {
+  assert(wasm_u16x8_extract_lane(v, 0) == l0);
+  assert(wasm_u16x8_extract_lane(v, 1) == l1);
+  assert(wasm_u16x8_extract_lane(v, 2) == l2);
+  assert(wasm_u16x8_extract_lane(v, 3) == l3);
+  assert(wasm_u16x8_extract_lane(v, 4) == l4);
+  assert(wasm_u16x8_extract_lane(v, 5) == l5);
+  assert(wasm_u16x8_extract_lane(v, 6) == l6);
+  assert(wasm_u16x8_extract_lane(v, 7) == l7);
+}
+
+void assert_lanes_eq_float(v128_t v,
+                           float l0,
+                           float l1,
+                           float l2,
+                           float l3,
+                           float l4,
+                           float l5,
+                           float l6,
+                           float l7) {
+  assert(wasm_f16x8_extract_lane(v, 0) == l0);
+  assert(wasm_f16x8_extract_lane(v, 1) == l1);
+  assert(wasm_f16x8_extract_lane(v, 2) == l2);
+  assert(wasm_f16x8_extract_lane(v, 3) == l3);
+  assert(wasm_f16x8_extract_lane(v, 4) == l4);
+  assert(wasm_f16x8_extract_lane(v, 5) == l5);
+  assert(wasm_f16x8_extract_lane(v, 6) == l6);
+  assert(wasm_f16x8_extract_lane(v, 7) == l7);
+}
+
+void assert_all_lanes_eq(v128_t v, float value) {
+  assert_lanes_eq_float(
+    v, value, value, value, value, value, value, value, value);
+}
+
+void assert_all_lanes_eq_uint16(v128_t v, uint16_t value) {
+  assert_lanes_eq_uint16(
+    v, value, value, value, value, value, value, value, value);
+}
+
+int main() {
+  v128_t a, b, c;
+  a = wasm_f16x8_splat(2.0f);
+  assert_all_lanes_eq(a, 2.0f);
+
+  a = wasm_f16x8_splat(1.0f);
+  a = wasm_f16x8_replace_lane(a, 0, 99.0f);
+  a = wasm_f16x8_replace_lane(a, 7, 99.0f);
+  assert_lanes_eq_float(a, 99.0f, 1.0f, 1.0f, 1.0f, 1.0f, 1.0f, 1.0f, 99.0f);
+
+  a = wasm_f16x8_abs(wasm_f16x8_splat(-2.0f));
+  assert_all_lanes_eq(a, 2.0f);
+
+  a = wasm_f16x8_neg(wasm_f16x8_splat(2.0f));
+  assert_all_lanes_eq(a, -2.0f);
+
+  a = wasm_f16x8_sqrt(wasm_f16x8_splat(4.0f));
+  assert_all_lanes_eq(a, 2.0f);
+
+  a = wasm_f16x8_ceil(wasm_f16x8_splat(1.2f));
+  assert_all_lanes_eq(a, 2.0f);
+
+  a = wasm_f16x8_floor(wasm_f16x8_splat(1.2f));
+  assert_all_lanes_eq(a, 1.0f);
+
+  a = wasm_f16x8_trunc(wasm_f16x8_splat(1.2f));
+  assert_all_lanes_eq(a, 1.0f);
+
+  a = wasm_f16x8_nearest(wasm_f16x8_splat(1.5f));
+  assert_all_lanes_eq(a, 2.0f);
+
+  a = wasm_f16x8_eq(wasm_f16x8_splat(2.0f), wasm_f16x8_splat(3.0f));
```

Comment:
It may be worth covering all the interesting cases for the comparisons to ensure we are getting the right comparison at runtime.